src/Migration/ErrorResolution/MigrationErrorResolutionContext.php

```diff
@@ -0,0 +1,78 @@
+<?php declare(strict_types=1);
+/*
+ * (c) shopware AG <info@shopware.com>
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace SwagMigrationAssistant\Migration\ErrorResolution;
+
+use Shopware\Core\Framework\Context;
+use Shopware\Core\Framework\Log\Package;
+
+/**
+ * @final
+ */
+#[Package('fundamentals@after-sales')]
+class MigrationErrorResolutionContext
+{
+    /**
+     * @param array<int|string, array<int|string, mixed>> $data
+     * @param array<string, list<MigrationFix>> $fixes
+     */
+    public function __construct(
```

Comment:
see above, in general if the class is already marked with `@final` (so no inheritance from it / decoration that might need to call the constructor) and if it is only constructed by us / the dependency injection (DI) container, then I would also always add `@internal` to the constructor just to be explicit that the constructor can change.
This is not need if the whole class is already marked `@internal`